### PR TITLE
Knex: Add tslint.json and fix issues, Fix dropTableIfExists to be chainable

### PR DESCRIPTION
**Pull request by Nospamas**
Please fill in this template.

- [√] Make your PR against the `master` branch.
- [√] Use a meaningful title for the pull request. Include the name of the package modified.
- [√] Test the change in your own code. (Compile and run.)
- [√] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [√] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [√] Run `tsc` without errors.
- [√] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [√] Provide a URL to documentation or source code which provides context for the suggested changes: http://knexjs.org/#Schema-dropTableIfExists
- [√] Increase the version number in the header if appropriate. **Added Version**
- [√] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by tkqubo** [comment]
:+1:

**Comment by tkqubo** [comment]
As commented in the code, later `on` could be extracted in some way :-)

File: knex/index.d.ts [hunk 1]
```diff
@@ -251,10 +259,12 @@ declare namespace Knex {
...
 
+    type UnionFunction = (this: QueryBuilder) => void;
+
     interface Union {
-        (callback: Function, wrap?: boolean): QueryBuilder;
-        (callbacks: Function[], wrap?: boolean): QueryBuilder;
-        (...callbacks: Function[]): QueryBuilder;
+        (callback: UnionFunction | UnionFunction[], wrap?: boolean): QueryBuilder;
+        (...callbacks: UnionFunction[]): QueryBuilder;
+        // no way to specifiy this siganiture in the current typescript version
```
Comment (minestarks) [inline knex/index.d.ts]: What do you mean by this comment?

**Review by minestarks — CHANGES_REQUESTED** [review]

File: knex/index.d.ts (hunk 1 above)
Comment (Nospamas) [inline knex/index.d.ts]: the comment is a note above:

// (...callbacks: Function[], wrap?: boolean): QueryInterface;

Which allows the specification of an arbitrary number of callbacks followed by an optional parameter to wrap them. Both spread operator arguments and optional arguments must be specified last, which as far as I know is not possible with typescript.

Docs for this function http://knexjs.org/#Builder-union

Comment (minestarks) [inline knex/index.d.ts]: Got it, thanks. Could you fix the typos in the comment and clarify that it refers to the  line below?

File: knex/index.d.ts [hunk 2]
```diff
@@ -23,7 +22,8 @@ interface Knex extends Knex.QueryInterface {
 
     raw: Knex.RawBuilder;
     transaction: <R>(transactionScope: ((trx: Knex.Transaction) => void)) => Promise<any>;
```
Comment (minestarks) [inline knex/index.d.ts]: This isn't your change, but it caught my attention -- this ```<R>``` is unnecessary and can be removed.

**Review by minestarks — CHANGES_REQUESTED** [review]

File: knex/index.d.ts (hunk 1 above)
Comment (Nospamas) [inline knex/index.d.ts]: Sure thing!

**Comment by Nospamas** [comment]
Clarified comment (if the wording was questionable to you it will be to others) and removed the unnecessary generic type.

**Review by minestarks — APPROVED** [review]

**Comment by minestarks** [comment]
@Nospamas thank you! Looks like you got some linter errors in tests. Once those are clean I'll merge.

**Comment by Nospamas** [comment]
Lint of tests updated. Looks like the default lint task doesn't hit non d.ts files when this repo was forked :)

**Comment by mhegazy** [comment]
looks like this needs a rebase from master.

**Comment by Nospamas** [comment]
I'm not hugely familiar with the rebase process and the history combined with the merge on the branch seems to be making it tough.

Any advice on this? Am I better off just re-making the repo given the age of this issue and when I originally forked it?

**Comment by paulvanbrenk** [comment]
I agree remaking the fix on a clean copy seems to right way forward here.

**Pull request closed without merging**